### PR TITLE
CE-3485

### PR DESCRIPTION
**Pull request by lukaszkonieczny**
## Links

* https://wikia-inc.atlassian.net/browse/CE-3484
* https://wikia-inc.atlassian.net/browse/CE-3485
* https://wikia-inc.atlassian.net/browse/CE-3471 (parent ticket)

## Description

When user scroll down a page over below given offset, we show banner on the bottom with feedback request. We have 6 variations of feedback form.

## Reviewers

@Wikia/x-wing @Wikia/community-engineering

### CODE REVIEW

File: front/main/app/components/bottom-banner.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import Ember from 'ember';
+import {getDomain} from '../utils/domain';
+
+export default Ember.Component.extend({
+	classNames: ['bottom-banner'],
+	classNameBindings: ['loaded', 'dismissed'],
+	dismissed: false,
+	loaded: false,
+
+	setCookie(cookieName, expires) {
```
Comment (rogatty) [inline front/main/app/components/bottom-banner.js]: Please add JSDoc.

File: front/main/app/components/feedback-form.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	textButtons: true,
+	emotButtons: false,
+	iconButtons: false,
+	displayQuestion: true,
+	displayInput: false,
+	displayThanks: false,
+	shouldDisplay: Ember.$.cookie('feedback-form'),
+	init() {
+		this._super(...arguments);
+		this.setVariation(this.get('variationId'));
```
Comment (rogatty) [inline front/main/app/components/feedback-form.js]: How about using this instead:
```
variation: Ember.computed(function () {
    return variations[this.variationId];
})
```
and in the template:
```
<p>{{variation.message}}</p>
{{#if (equal variation.buttons 'text')}}
    ...
{{else if (equal variation.buttons 'emot')}}
    ...
{{else}}
    ...
```
?

File: front/main/app/components/feedback-form.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+		if (!Ember.$.cookie(cookieName)) {
+			Ember.run.scheduleOnce('afterRender', this, () => {
+				const pageHeight = document.getElementsByClassName('wiki-container')[0].offsetHeight;
+
+				this.set('bannerOffset', Math.floor(pageHeight * offsetLimit));
+				Ember.$(window).on('scroll.feedbackForm', this.checkOffsetPosition.bind(this));
```
Comment (rogatty) [inline front/main/app/components/feedback-form.js]: Would this work?
```
Ember.$(window).on('scroll.feedbackForm', () => this.checkOffsetPosition);
```

File: front/main/app/components/feedback-form.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+		const scrollY = window.scrollY,
+			direction = scrollY >= this.get('lastOffset') ? 0 : 1;
+
+		this.set('lastOffset', scrollY);
+
+		Ember.run.debounce({}, () => {
+			if ((this.get('firstDisplay') || scrollY > this.get('bannerOffset')) && direction) {
+				this.set('loaded', true);
+				this.set('dismissed', false);
+				this.set('firstDisplay', true);
```
Comment (rogatty) [inline front/main/app/components/feedback-form.js]: Use `this.setProperties({ ... });`

File: front/main/app/components/feedback-form.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+	},
+	resetVariations() {
+		this.setProperties({
+			textButtons: false,
+			emotButtons: false,
+			iconButtons: false
+		});
+	},
+	dismissBanner(timeout) {
+		Ember.$(window).off('scroll.feedbackForm');
```
Comment (rogatty) [inline front/main/app/components/feedback-form.js]: Let's make sure that this is always unbinded. E.g. by using `willDestroyElement()`.

File: front/main/app/components/feedback-form.js (hunk 3 above)
Comment (lukaszkonieczny) [inline front/main/app/components/feedback-form.js]: This works :)

`Ember.$(window).on('scroll.feedbackForm', () => this.checkOffsetPosition());`

**Comment by rogatty** [comment]
LGTM

**Pull request merged**